packages/codegen-ui-react/lib/forms/component-helper.ts

```diff
@@ -0,0 +1,67 @@
+/*
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+import { FieldConfigMetadata } from '@aws-amplify/codegen-ui';
+import { factory, SyntaxKind } from 'typescript';
+
+export const ControlledComponents = ['StepperField', 'SliderField', 'SelectField'];
+
+/**
+ * given the component returns true if the component is a controlled component
+ *
+ * @param componentType
+ * @returns
+ */
+export const isControlledComponent = (componentType: string): boolean => ControlledComponents.includes(componentType);
+
+/**
+ *
+ * @param stateName name of the state name to render for the default value
+ * @param { dataType } the dataType
+ * @returns
+ */
+export const renderDefaultValueAttribute = (stateName: string, { dataType }: FieldConfigMetadata) => {
+  let expression = factory.createJsxExpression(undefined, factory.createIdentifier(stateName));
+
+  if (dataType && dataType === 'AWSTimestamp') {
+    expression = factory.createJsxExpression(
+      undefined,
+      factory.createBinaryExpression(
+        factory.createIdentifier(stateName),
+        factory.createToken(SyntaxKind.AmpersandAmpersandToken),
+        factory.createCallExpression(factory.createIdentifier('getDateTimeLocal'), undefined, [
+          factory.createIdentifier(stateName),
+        ]),
+      ),
+    );
+  }
+
+  if (dataType && dataType === 'AWSDateTime') {
+    expression = factory.createJsxExpression(
+      undefined,
+      factory.createCallExpression(
+        factory.createPropertyAccessExpression(
+          factory.createIdentifier(stateName),
+          factory.createIdentifier('replace'),
+        ),
+        undefined,
+        [factory.createStringLiteral('Z'), factory.createStringLiteral('')],
+      ),
+    );
+  }
+
+  return factory.createJsxAttribute(factory.createIdentifier('defaultValue'), expression);
```

Comment:
I think the defaultValue attribute name is going to change per component type. I think we need a map of componenttype to default value attribute.

For example,

Reply:
adding a map in the next revision